src/repr/src/adt/datetime.rs

```diff
@@ -349,6 +322,10 @@ impl DateTimeFieldValue {
     pub fn new(unit: i64, fraction: i64) -> Self {
         DateTimeFieldValue { unit, fraction }
     }
+
+    /// How much padding is added to the fractional portion to achieve a given precision.
+    /// e.g. with the current precision `.5` is represented as `500_000_000`.
+    const FRACTIONAL_DIGIT_PRECISION: i64 = 1_000_000_000;
```

Comment:
Could this be replaced by `convert_date_time_unit`?

Reply:
To be honest, I'm not exactly sure how to go about that. This is used to convert between a decimal part of a float and an integer. For example 0.123456789 to 1234556789. It's not exactly about converting between different time/date units.